doc/gen_support_matrix_html.rs

```diff
@@ -0,0 +1,426 @@
+    // TODO: read the order from the YAML file instead?
+    boards.sort_unstable_by_key(|b| b.name.clone());
+
+    let mut env = Environment::new();
+    env.add_template("matrix", TABLE_TEMPLATE).unwrap();
+    env.add_template("matrix_key", KEY_TEMPLATE).unwrap();
+
+    let tmpl = env.get_template("matrix").unwrap();
+    let matrix_html = tmpl.render(context!(matrix => matrix, boards => boards)).unwrap();
+
+    let tmpl = env.get_template("matrix_key").unwrap();
+    let key_html = tmpl.render(context!(matrix => matrix, boards => boards)).unwrap();
+
+    // NOTE: We may want to return the table and its key separately later
+    Ok(format!("{matrix_html}{key_html}\n"))
+}
+
+#[derive(Debug, thiserror::Error, Diagnostic)]
+enum Error {
+    #[error("could not find file `{path}`")]
+    InputFile {
+        path: PathBuf,
+        source: io::Error,
+    },
+    #[error("could not parse YAML file `{path}`")]
+    Parsing {
+        path: PathBuf,
+        #[source_code]
+        src: String,
+        #[label = "Syntax error"]
+        err_span: miette::SourceSpan,
+        source: serde_yaml::Error,
+    },
+    #[error("invalid chip name `{found}` for board `{board}`")]
+    InvalidChipName {
+        found: String,
+        board: String,
+    },
+    #[error("invalid functionality name `{found}` for board `{board}`")]
+    InvalidFunctionalityNameBoard {
+        found: String,
+        board: String,
+    },
+    #[error("invalid functionality name `{found}` for chip `{chip}`")]
+    InvalidFunctionalityNameChip {
+        found: String,
+        chip: String,
+    },
+    #[error("invalid support key name `{found}` for functionality `{functionality}` for board `{board}`")]
+    InvalidSupportKeyNameBoard {
+        found: String,
+        functionality: String,
+        board: String,
+    },
+    #[error("invalid support key name `{found}` for functionality `{functionality}` for chip `{chip}`")]
+    InvalidSupportKeyNameChip {
+        found: String,
+        functionality: String,
+        chip: String,
+    },
+    #[error("missing support info on board `{board}` or chip `{chip}` regarding functionality `{functionality}`")]
+    MissingSupportInfo {
+        board: String,
+        chip: String,
+        functionality: String,
+    },
+    #[error("could not write the output HTML file `{path}`")]
+    WritingOutputFile {
+        path: PathBuf,
+        source: io::Error,
+    },
+    #[error("could not read existing output HTML file `{path}`")]
+    ReadingExistingFile {
+        path: PathBuf,
+        source: io::Error,
+    },
+    #[error("existing HTML file `{path}` is not up to date")]
+    ExistingHtmlNotUpToDate {
+        path: PathBuf,
+    },
+}
+
+mod schema {
+    use std::collections::HashMap;
+
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct Matrix {
+        pub support_keys: HashMap<String, SupportKeyInfo>,
```

Comment:
It's one dependency more, but may I suggest using `IndexMap` here? That way the user has some say about the order.